cmake/modules/FindCrossGUID.cmake

```diff
@@ -2,30 +2,24 @@ if(ENABLE_INTERNAL_CROSSGUID)
   include(ExternalProject)
   include(cmake/scripts/common/ModuleHelpers.cmake)
 
-  get_archive_name(crossguid)
+  set(MODULE_LC crossguid)
 
-  # allow user to override the download URL with a local tarball
-  # needed for offline build envs
-  if(CROSSGUID_URL)
-    get_filename_component(CROSSGUID_URL "${CROSSGUID_URL}" ABSOLUTE)
-  else()
-    set(CROSSGUID_URL http://mirrors.kodi.tv/build-deps/sources/${CROSSGUID_ARCHIVE})
-  endif()
-  if(VERBOSE)
-    message(STATUS "CROSSGUID_URL: ${CROSSGUID_URL}")
-  endif()
+  SETUP_BUILD_VARS()
 
   if(APPLE)
     set(EXTRA_ARGS "-DCMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES}")
   endif()
 
   set(CROSSGUID_LIBRARY ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/lib/libcrossguid.a)
   set(CROSSGUID_INCLUDE_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/include)
-  externalproject_add(crossguid
-                      URL ${CROSSGUID_URL}
-                      URL_HASH ${CROSSGUID_HASH}
+  set(CROSSGUID_VER ${${MODULE}_VER})
+
+  externalproject_add(${MODULE_LC}
+                      URL ${${MODULE}_URL}
+                      URL_HASH ${${MODULE}_HASH}
                       DOWNLOAD_DIR ${TARBALL_DIR}
-                      PREFIX ${CORE_BUILD_DIR}/crossguid
+                      DOWNLOAD_NAME ${${MODULE}_ARCHIVE}
+                      PREFIX ${CORE_BUILD_DIR}/${MODULE_LC}
```

Comment:
all those indirect references like ${${MODULE}_XXX} are unnecessarily hard to read.
Why not use CROSSGUID_XXX instead?

Reply:
The end goal is to template everything out. A single helper function/macro. So the less "hardcoded" names, the easier that is to do.

Reply:
An example of some of the testing/WIP stuff i did for other dependencies can be spotted here for an idea about the templating

https://github.com/fuzzard/xbmc/commits/cmake_python

Just keep in mind, its just a dump of stuff i was playing with. Theres a few depends (libass, python3 + modules, cdio, etc) that i used. Essentially the format i was trying was pull out the build stuff from the FindModules into their own folder, existing FindModules then call the buildmodule if desired, and the intent would be to just glob everything else in and work through their dependences. its EXTREMELY wip, and no where near that state, and im not sure i'll even keep going that path as yet, but was just something to play with.